packages/features/form-builder/FormBuilder.tsx

```diff
@@ -266,8 +266,7 @@ export const FormBuilder = function FormBuilder({
           {fields.map((field, index) => {
             const fieldType = FieldTypesMap[field.type];
 
-            // Hidden fields can't be required
-            const isRequired = field.required && !field.hidden;
+            const isRequired = field.required;
```

Comment:
No need to show the field as optional now. As mentioned by @Jaibles we would not be showing 'Optional' badge if field is hidden. There would just be Hidden badge.